docs/get-started/cluster-scale-out.md

```diff
@@ -0,0 +1,225 @@
+# Cluster Scale Out
+
+Cluster expansion allows the user to add new machine nodes to an existing cluster and to redistribute data. This document introduces how to horizontally scale out a nebula cluster. Horizontally scaling out nebula means scaling nebula out by adding nodes. When scaling out a cluster, the data is scaled out sequentially according to the values of replica value.
+
+## Prerequisites
+
+- The current nebula system must be installed in a cluster mode, not a single-node mode.
+- The new nodes are available.
+
+## Cluster Expansion Workflow
+
+### Connect to Nebula
+
+First check your IP address and configure **nebula-metad.conf** using the following command.
+
+```
+$ ip addr
+$ cd /usr/local/nebula/etc
+$ vi nebula-metad.conf
+
+########## networking ##########
+# Local ip
+--local_ip={your local ip}
+```
+
+When configuration is done, try to connect to nebula using the following command.
+
+```
+$ scripts/nebula.service start all
+
+[WARN] The maximum files allowed to open might be too few: 1024
+[INFO] Starting nebula-metad...
+[INFO] Done
+[INFO] Starting nebula-graphd...
+[INFO] Done
+[INFO] Starting nebula-storaged...
+[INFO] Done
+
+$ scripts/nebula.service status all
+
+[INFO] nebula-metad: Running as 21935, Listening on 45500
+[INFO] nebula-graphd: Running as 22024, Listening on 3699
+[INFO] nebula-storaged: Running as 22042, Listening on 44500
+```
+
+The above information indicates that nebula has been connected successfully. Some users may receive a warning `The maximum files allowed to open might be too few: 1024`, change the `/etc/security/limits.conf` file using the following command.
+
+```
+ulimit -v
+```
+
+## Add New Nodes to Cluster
+
+Assuming the original cluster has 7 hosts with 3 partitions, 3 replicas. Let's scale the cluster to 9 hosts (one partition one host). In this document, we only scale out storage service, therefore only start multiple storages using the following command.
+
+```
+$ cd /usr/local/nebula/etc
+$ cp nebula-storaged.conf nebula-storaged2.conf
+$ cp nebula-storaged.conf nebula-storaged3.conf
+$ cp nebula-storaged.conf nebula-storaged4.conf
+$ cp nebula-storaged.conf nebula-storaged5.conf
+$ cp nebula-storaged.conf nebula-storaged6.conf
+$ cp nebula-storaged.conf nebula-storaged7.conf
+$ cp nebula-storaged.conf nebula-storaged8.conf
+$ cp nebula-storaged.conf nebula-storaged9.conf
+
+```
+
+Configure the storage files as follows, take **nebula-storaged2.conf** as example.
+
+```
+# The file to host the process id
+--pid_file=pids/nebula-storaged2.pid
+# Storage daemon listening port
+--port=34510
+# HTTP service port
+--ws_http_port=12100
+# HTTP2 service port
+--ws_h2_port=12012
+# One path per instance, if --engine_type is `rocksdb'
+--data_path=data/storage2
+```
+
+**_Note_:** Make sure the ports are available when configuring with the command `netstat -anp|grep {port number}`.
```

Comment:
PS: the storage port can't be consecutive, for example, if storage1 port is 10001, storage port is 10002, it would make the service abnormal.